src/plugins/presentation_util/public/components/saved_object_save_modal_dashboard.tsx

```diff
@@ -30,7 +30,7 @@ export interface SaveModalDashboardProps {
   documentInfo: SaveModalDocumentInfo;
   objectType: string;
   onClose: () => void;
-  onSave: (props: OnSaveProps & { dashboardId: string | null }) => void;
+  onSave: (props: OnSaveProps & { dashboardId: string | null; addToLibrary: boolean }) => void;
```

Comment:
We have a fundamental problem at the moment with "prop drilling", where, rather than allowing a component to own its state, we place that state in a consuming component... this creates a dependent, complex relationship between the consumer and the component that becomes impossible to break.

A rule of thumb: allow a component to own its state, and have the consuming component provide a handler-- e.g. `onClick`, `onAdd`, etc-- and provide a default or make it required.  This more closely follows the `DOM` and makes your component contract that much clearer to other consumers.

I'll file a PR against this branch with an example.